### PR TITLE
Port To Gtk 3

### PR DESCRIPTION
**Pull request by Creatune**
I ported a bit by following the guide, but the activity seems to give an error. I will check it out and try to solve it :).

### CODE REVIEW

**Comment by Creatune** [comment]
I am figuring out how to use the pygtk module. Please let me know if there is anything else that I can add.

File: JAMClock.py [hunk 1]
```diff
@@ -19,7 +19,8 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
-import os, gtk, pygtk, gobject, sys, socket, pygame
+from gi.repository import Gtk
+import os, pygtk, gobject, sys, socket, pygame
```
Comment (tchx84) [inline JAMClock.py]: You should also do `from gi.repository import GObject`. As it is, you import `gobject` but then reference `GObject`. It shouldn't work.

**Review by tchx84 — CHANGES_REQUESTED** [review]

File: JAMClock.py (hunk 1 above)
Comment (Creatune) [reply JAMClock.py]: Yikes! My bad. I will make the changes immediately.

File: activity/activity.info [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [Activity]
 name = JAMClock
 activity_version = 2
-service_name = org.laptop.JAMClock
+bundle_id = org.sugarlabs.JAMClock
```
Comment (quozl) [inline activity/activity.info]: Why change the name from org.laptop to org.sugarlabs?  This will break resuming any journal entries after the activity is resumed.

File: activity/activity.info [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
 icon = JAMClock
 show_launcher = yes
-class = JAMClock.JAMClock
+exec = sugar-activity3 JAMClock.JAMClock
```
Comment (quozl) [inline activity/activity.info]: Why change to Python 3 wrapper when porting to GTK 3?  Are you also porting to Python 3?  If so, please rename the pull request and change the commit messages.

File: JAMClock.py [hunk 4]
```diff
@@ -185,7 +188,7 @@ def __init__(self):
 		self.connect("embedded", self.embed_event)
 
 	       	os.putenv('SDL_WINDOWID', str(self.eventbox.socket.get_id()))
-		gobject.idle_add(self.get_run_game)
+		GObject.idle_add(self.get_run_game)
```
Comment (quozl) [inline JAMClock.py]: GObject.idle_add is an alias for GLib.idle_add, and we get depreciation warnings for it.

File: JAMClock.py [hunk 5]
```diff
@@ -173,9 +176,9 @@ def salir(self, widget):
 		pygame.quit()
 		sys.exit()
 
-class gtkplug(gtk.Plug):
+class gtkplug(Gtk.Plug):
 	def __init__(self):
-		gtk.Plug.__init__(self, 0L)
+		GObject.GObject.__init__(self, 0L)
```
Comment (quozl) [inline JAMClock.py]: Why this change; i.e. how to ensure `Gtk.Plug.__init__` is called?

File: JAMClock.py [hunk 6]
```diff
@@ -144,19 +147,19 @@ def set_focus(self, container, widget):
 # -----------------------------------------------
 # 	******** El Juego en gtk ********
 # -----------------------------------------------
-class VentanaGTK(gtk.Window):
+class VentanaGTK(Gtk.Window):
 	def __init__(self):
-		gtk.Window.__init__(self, gtk.WINDOW_TOPLEVEL)
+		GObject.GObject.__init__(self, Gtk.WindowType.TOPLEVEL)
```
Comment (quozl) [inline JAMClock.py]: How to ensure `Gtk.Window.__init__` is called?

File: JAMClock.py [hunk 7]
```diff
@@ -51,23 +54,23 @@ def get_run_game(self):
...
-		lambda w: gtk.main_quit()
+		lambda w: Gtk.main_quit()
 		sys.exit()
 
-class PygameCanvas(gtk.EventBox):
+class PygameCanvas(Gtk.EventBox):
 	def __init__(self):
-		gtk.EventBox.__init__(self)
-		self.set_flags(gtk.CAN_FOCUS)
+		GObject.GObject.__init__(self)
```
Comment (quozl) [inline JAMClock.py]: How to ensure `Gtk.EventBox.__init__` is called?

File: JAMClock.py [hunk 8]
```diff
@@ -34,14 +37,14 @@ def __init__(self, handle):
...
-		self.add_events(gtk.gdk.ALL_EVENTS_MASK)
+		self.add_events(Gdk.EventMask.ALL_EVENTS_MASK)
 		self.connect("destroy", self.salir)
 
 		self.show_all()
 		self.realize()
 
 	       	os.putenv('SDL_WINDOWID', str(self.eventbox.socket.get_id()))
-		gobject.idle_add(self.get_run_game)
+		GObject.idle_add(self.get_run_game)
```
Comment (quozl) [inline JAMClock.py]: GObject.idle_add causes deprecation warnings.

**Review by quozl — CHANGES_REQUESTED** [review]
Please test before commit, thanks.

Also, not sure if you are doing a port to GTK 3 at the same time as a port to Python 3.

**Comment by Creatune** [comment]
But due to the latest commit (14fe4f4), I have observed that the following error has repeated:
https://asciinema.org/a/jcDlDCFifmAkma1YtTgwOWci6
How can I fix this?

File: activity/activity.info (hunk 3 above)
Comment (Creatune) [reply activity/activity.info]: I was planning to port to python 3 first but then decided not to, I probably forgot to revert this change. My bad.

**Comment by Creatune** [comment]
I also ported this activity from GObject to GLib. Is that required? Or did I do something extra?

File: activity/activity.info [hunk 9]
```diff
@@ -1,7 +1,7 @@
 [Activity]
 name = JAMClock
 activity_version = 2
-bundle_id = org.sugarlabs.JAMClock
+bundle_id = org.laptop.JAMClock
 icon = JAMClock
 show_launcher = yes
-exec = sugar-activity3 JAMClock.JAMClock
+exec = JAMClock.JAMClock
```
Comment (quozl) [inline activity/activity.info]: `exec` must contain either sugar-activity or sugar-activity3; you've removed both, and this causes the traceback you observed;

```log
Traceback (most recent call last):
File "/usr/lib/python2.7/dist-packages/sugar3/activity/activityinstance.py", line 175, in main
splitted_module = activity_class.rsplit('.', 1)
AttributeError: 'NoneType' object has no attribute 'rsplit'   
```

**Comment by Creatune** [comment]
I am so sorry. I am yet not clear with the concept and have school stuff to catch up with. I will get back to this, even if GCI ends.

**Comment by quozl** [comment]
Thanks.  Good start made.  I'll close this pull request now.  We can either re-open it later, or you can open another one.  Let us know.
